### ISSUE TITLE
Be able to upgrade core resources in an existing project on editor upgrades

### ISSUE THREAD

**Issue by shaddockh** [AtomicGameEngine/AtomicGameEngine#757]
Based upon discussions with both @mattbenic  and @JoshEngebretson, it seems that there should be a way to upgrade current resources in a project when a new version of the editor drops.  One main culprit of this is to be able to bring in the latest *.d.ts files on each new update.  I'm thinking that this could be delegated to the extension subsystem.  What I'm thinking is that the editor can execute a method on the extension ServiceLocator:
```
ServiceLocator.isUpgradeAvailable({editor version});
```
If any extensions subscribed to this event return ```true``` then either an upgrade menu item appears or a dialog box pops up stating : "Project Upgrades Available.  Do you want to apply them?"

If the answer is yes, then the editor calls
```
ServiceLocator.applyUpgrades
```
And then it is up to each extension to upgrade the relevant files as necessary.  For instance the TypeScript extension could have the latest *.d.ts files somewhere and it could overwrite the files in ```ProjectRoot/typings/Atomic/*```

**Comment by mattbenic** [comment]
Yeah, good thinking. There's already a project version field in the .atomic file. Would you re-prompt for upgrades on each open, or flag somewhere that the prompt has happened? It seems to me that should be in a user specific file rather than a project specific one. Perhaps a <projectname>.userprefs or something similar in the root of the project, that can be ignored by source control.
